File: pilhas-filas/csharp/stack2-test/Stack2.cs
using System;
using System.Collections.Generic;

public class Stack2<T>
{
    private List<T> list;

    public Stack2()
    {
        this.list = new List<T>();
    }

    public bool IsEmpty()
    {
        return this.list.Count == 0;
    }

    public void Push(T item)
    {
        this.list.Add(item);
    }

    public T Pop()
    {
        if (this.IsEmpty())
        {
            throw new InvalidOperationException("Stack is empty");
        }
        T item = this.list[this.list.Count - 1];
        this.list.RemoveAt(this.list.Count - 1);
        return item;
    }

    public T Peek()
    {
        if (this.IsEmpty())
        {
            throw new InvalidOperationException("Stack is empty");
        }
        return this.list[this.list.Count - 1];
    }

    public int Count
    {
        get 
        {
            return this.list.Count;
        }
    }
}

File: pilhas-filas/python/stack2.py
class Stack2:
    def __init__(self):
        self.list = []

    def is_empty(self):
        return len(self.list) == 0

    def push(self, item):
        self.list.append(item)

    def pop(self):
        if self.is_empty():
            raise Exception("Stack is empty")
        return self.list.pop()

    def peek(self):
        if self.is_empty():
            raise Exception("Stack is empty")
        return self.list[-1]

    def count(self):
        return len(self.list)

    def clear(self):
        self.list = []
